### PR TITLE
[Staleness] Handle Overlay Plots in Stacked Plots and removing LAD Tables from LAD Table Sets

### LINKED ISSUE #6265: Staleness doesn't work for overlay plots embedded in stacked plots
#### Summary

Staleness doesn't show on overlay plots embedded in stacked plots

#### Expected vs Current Behavior

Staleness indicator should show

#### Steps to Reproduce

1. Create an overlay plot
2. Create an SWG with staleness option ON
3. Drag SWG into the overlay plot
4. Create a stacked plot
5. Drag the original SWG and the overlay plot into the stacked plot
6. Watch it and observe that staleness border shows up for SWG but not the overlay plot

#### Environment

* Open MCT Version:  2.1.6-SNAPSHOT
* Deployment Type:  testathon
* OS: macos
* Browser: chrome

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [ ] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

#### Additional Information

Pasted Graphic

### PR DESCRIPTION
**Pull request by jvigliotta**
Closes #6265
And addresses: https://github.com/nasa/openmct/issues/6109#issuecomment-1414231882 (LAD Table Set part, other is this issue)

### Describe your changes:

If an item is added to a Stacked Plot that is NOT a telemetry producing item, then we'll watch it's composition for staleness. Also, when removing a LAD Table from a LAD Table Set, we will stop observing staleness for it's telemetry. And a typo fix. ALSO, also, fixing the `isStale` method of the example SWG Staleness Provider.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes? (forthcoming)
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Review by ozyx — CHANGES_REQUESTED** [review]
Looking good! Smoke tested and seems to work well. I left some suggestions.

Could we add an e2e test or two to cover these issues?

**Comment by jvigliotta** [comment]
@ozyx regarding tests, will be adding a staleness test suite this week. Thanks for the review!

File: src/plugins/plot/stackedPlot/StackedPlotItem.vue [hunk 1]
```diff
@@ -230,6 +257,46 @@ export default {
...
+                    this.handleStaleness(keyString, stalenessResponse);
+                }
+            });
+            const stalenessSubscription = this.openmct.telemetry.subscribeToStaleness(object, (stalenessResponse) => {
+                this.handleStaleness(keyString, stalenessResponse);
+            });
+
+            this.stalenessSubscription[keyString].unsubscribe = stalenessSubscription;
+        },
+        unwatchStaleness(object) {
```
[the remark this answers is not in the document]
Comment (jvigliotta) [reply src/plugins/plot/stackedPlot/StackedPlotItem.vue:276]: Gotcha, doing it above as well in `watchStaleness`

File: src/plugins/plot/stackedPlot/StackedPlotItem.vue [hunk 2]
```diff
@@ -328,6 +395,20 @@ export default {
...
                 return this.childObject;
             }
+        },
+        destroyStalenessListeners() {
+            this.triggerUnsubscribeFromStaleness();
+
+            if (this.composition) {
+                this.composition.off('add', this.watchStaleness);
+                this.composition.off('remove', this.unwatchStaleness);
+                this.component = null;
```
[the remark this answers is not in the document]
Comment (jvigliotta) [reply src/plugins/plot/stackedPlot/StackedPlotItem.vue:405]: ah... this was supposed to be compostion... :) good catch!

File: src/plugins/LADTable/components/LadTableSet.vue [hunk 3]
```diff
@@ -48,7 +48,7 @@
                 </tr>
                 <lad-row
                     v-for="ladRow in ladTelemetryObjects[ladTable.key]"
-                    :key="ladRow.key"
+                    :key="combineKeys(ladTable.key, ladRow.key)"
```
[the remark this answers is not in the document]
Comment (jvigliotta) [reply src/plugins/LADTable/components/LadTableSet.vue:51]: Yes, you can only have one unique domainObject per lad table. Previously, it was NOT unique as you could have the same domainObject in two different lad tables.

**Review by ozyx — APPROVED** [review]
LGTM

**Comment by jvigliotta** [comment]
Issue for adding tests: https://github.com/nasa/openmct/issues/6152

**Pull request merged**